examples/extra_element_integer/random_cluster.cpp

```diff
@@ -0,0 +1,128 @@
+#include "libmesh/libmesh.h"
+#include "libmesh/mesh.h"
+#include "libmesh/exodusII_io.h"
+#include "libmesh/elem.h"
+#include "libmesh/equation_systems.h"
+#include "libmesh/linear_implicit_system.h"
+#include "libmesh/dof_map.h"
+#include "libmesh/numeric_vector.h"
+#include "libmesh/system.h"
+#include "libmesh/mesh_generation.h"
+
+#include <vector>
+#include <cstdlib>
+#include <ctime>
+
+using namespace libMesh;
+
+void CreateMesh(libMesh::Mesh & mesh, int nx, int ny){
+
+    libMesh::MeshTools::Generation::build_square(mesh,
+                                                 nx,
+                                                 ny,
+                                                 0.0, 10.0,
+                                                 0.0, 10.0,
+                                                 libMesh::QUAD4);
+}
+
+void CreateMesh(libMesh::Mesh & mesh, int nx, int ny, int nz ){
+    libMesh::MeshTools::Generation::build_cube(mesh,
+                                               nx,
+                                               ny,
+                                               nz,
+                                               0.0, 10.0,
+                                               0.0, 10.0,
+                                               0.0, 10.0);
+}
+
+void PopulateRandomIntegers(libMesh::Mesh & mesh, LinearImplicitSystem& system, const unsigned int &index ) {
+
+    DofMap& dof_map = system.get_dof_map();
+    for (const auto& elem : mesh.element_ptr_range()) {
+
+        std::vector<dof_id_type> dof_indices;
+        dof_map.dof_indices(elem, dof_indices);
+        int random_int = rand() %5;
+
+        system.solution->set(dof_indices[1], random_int);
+        elem->set_extra_integer(index, 5-random_int);
```

Comment:
We should NOT set the extra integer here.  The extra integer will hold the value of the cluster ID.  If we want this app to be a model for the real system, this step is a surrogate for doing real physics.  Determining the cluster ID is a different step and needs to be considered elsewhere.  It's fine, for now, if you still choose to set the cluster ID based mathematically on the random int, but it should be separated from this step.